Source/Engine/UI/GUI/Control.cs

```diff
@@ -310,7 +310,7 @@ public bool VisibleInHierarchy
         /// <summary>
         /// Gets the GUI window root control which contains that control (or null if not linked to any).
         /// </summary>
-        public virtual WindowRootControl RootWindow => _parent?.RootWindow;
+        public virtual WindowRootControl RootWindow => _parent?.RootWindow; // TODO: Why doesn't this just go "_root?.RootWindow" or something?
```

Comment:
Like this says, why doesn't it instantly jump to the `_root`? Is it so that any parent control can override this?

Reply:
Actually, it could be optimized as we cache `_root` when the controls hierarchy gets modified in `Parent` setter. Can you test it if it works fine with this tweak?

Reply:
Okay, will do!

Reply:
Okay, checked the code and tested it out. Works as expected